src/main/java/com/zaxxer/hikari/proxy/ConnectionProxy.java

```diff
@@ -147,24 +147,27 @@ public final void markCommitStateDirty()
       return statement;
    }
 
-   private final void closeOpenStatements()
+   private final boolean closeOpenStatements()
    {
       final int size = openStatements.size();
-      if (size > 0) {
-         for (int i = 0; i < size; i++) {
-            try {
-               final Statement statement = openStatements.get(i);
-               if (statement != null) {
-                  statement.close();
-               }
-            }
-            catch (SQLException e) {
-               checkException(e);
+      if (size <= 0) {
+         return true;
+      }
+      boolean success = true;
+      for (int i = 0; i < size; i++) {
+         try {
+            final Statement statement = openStatements.get(i);
+            if (statement != null) {
+               statement.close();
             }
          }
-
-         openStatements.clear();
+         catch (SQLException e) {
+            checkException(e);
+            success &= !poolEntry.evict;
+         }
       }
+      openStatements.clear();
+      return success;
    }
```

Comment:
you mean:
if (size <= 0) {
   return true;
}
I think returning early this way is more readable
this saves rest of method from indentation as well